src/util/paths.js

```diff
@@ -77,22 +80,79 @@ module.exports = {
    * @param {String} guideId - id of the guide.
    * @param {String} fileDataUrl - used for standalone assembly
    *
-   * @return {Promise} a Promise that will resolve to the
-   * path to the templates.json file for the current Interview.
+   * @return {Promise}
    */
-  getTemplatesPath ({ username, guideId, fileDataUrl }) {
+  async getTemplatesPath ({ username, guideId, fileDataUrl }) {
     const deferred = Q.defer()
     const guidesDir = config.get('GUIDES_DIR')
 
-    const templatesPath = fileDataUrl
-      ? path.join(this.normalizeFileDataUrl(fileDataUrl), 'templates.json')
-      : path.join(guidesDir, username, 'guides', `Guide${guideId}`, 'templates.json')
+    // get the root path, minus the templates.json
+    const rootPath = fileDataUrl
+      ? path.join(this.normalizeFileDataUrl(fileDataUrl))
+      : path.join(guidesDir, username, 'guides', `Guide${guideId}`)
+    const templatesPath = path.join(rootPath, 'templates.json')
 
-    deferred.resolve(templatesPath)
+    // catch missing templates.json -- should rarely happen
+    const exists = await files.pathExists(templatesPath)
+    if (!exists) {
+      try {
+        const guideFiles = await files.readDir({ path: rootPath })
+        const templateIds = await this.getTemplateIds(guideFiles)
+        await this.createTemplatesJSON(
+          templatesPath,
+          guideId,
+          templateIds
+        )
+      } catch (err) {
+        console.error(err)
+      }
+    }
 
+    deferred.resolve(templatesPath)
     return deferred.promise
   },
 
+  // return an array of templateIds
+  getTemplateIds (guideFiles) {
+    return new Promise((resolve, reject) => {
+      const templateFiles = guideFiles.filter(
+        filename =>
+          filename.startsWith('template') &&
+          filename.endsWith('.json') &&
+          filename !== 'templates.json'
+      )
+
+      let templateIds = templateFiles.map(fileName => {
+        const startIndex = fileName.indexOf('template') + 8
+        const endIndex = fileName.indexOf('.', startIndex)
+
+        return parseInt(fileName.substring(startIndex, endIndex))
+      })
+
+      templateIds = templateIds.filter((c, index) => {
```

Comment:
This is making sure that the values in the `templateIds` array are all numbers. 
eg. if there were a file called `templateXYZ.json`, it would be added to the `templateFiles` first on line 118. So, the `templateIds`  array would have an NaN value at 129. 
The second filter takes NaN value out of the `templateIds`